src/unix/mod.rs

```diff
@@ -0,0 +1,135 @@
+use crate::Error;

```

Comment:
As there's no other module inside of the `unix` directory, why not just moving this to `src/unix.rs`, if in the future we have another module just for unix I think it would make sense to put it inside of a `unix` directory, WDYT?

Reply:
yes, but windows implementation should have an Error too.

Reply:
Sure so we could
have Errors for each module, as long as they are documented I don't see why not!

Reply:
> yes, but windows implementation should have an Error too.

The error for windows only is making me doubt on `match` implementations, I mean, me as a client of this crate will haver to always write a path for the `GetAdaptersAddresses` error right?